app/controllers/v1/districts_controller.rb

```diff
@@ -0,0 +1,59 @@
+class V1::DistrictsController < V1::BaseController
+  def index
+    if address = params[:address].presence
+      results = here_coords(address, params[:city], params[:state], params[:zip])
+      coords = results[:coordinates]
+      district_info = coords ? mcommons_district(coords) : {}
+      
+      render json: { address:    results[:address_name],
+                     confidence: results[:confidence],
+                     coords: coords
+                   }.merge(district_info)
+    
+    elsif zip = params[:zip].presence
+      render json: district_info_for_zip(zip),
+              status: 200
+    else
+      render json: { error: 'zip code is required' }
+    end
+  end
+
+  private
+    def district_info_for_zip(zip)
+      targeted, district, city, state = nil
+      if zip_code = ZipCode.find_by(zip_code: zip)
+        city = zip_code.city
+        state = zip_code.state.abbrev
+        if zip_code.campaigns.include?(Campaign.first)
+          if zip_code.districts.count == 1
+            district = zip_code.districts.first
+            state = district.state.abbrev
+            district = district.district
+            targeted = true  
+          end
+        else
+          targeted = false
+        end
+      end
+      { targeted: targeted, district: district, city: city, state: state }
+    end
+
+    def mcommons_district(coords)
+      results = Integration::MobileCommons.district_from_coords(coords)
+      state = State.find_by(abbrev: results[:state])
```

Comment:
Not sure you need to go through state here - why not use the state provided by Mcommons?